pysmt/rewritings.py

```diff
@@ -832,3 +887,60 @@ def disjunctive_partition(formula):
                 to_process += cur.args()
             else:
                 yield cur
+
+
+def propagate_toplevel(formula, env=None, do_simplify=True, preserve_equivalence=True):
+    """ Propagates the toplevel definitions and returns an equivalent formula.
+    It considers two kinds of definitions:
+    1) variable = constant
+    2) variable = variable
+    """
+    if env is None:
+        import pysmt.environment
+        env = pysmt.environment.get_env()
+    mgr = env.formula_manager
+
+    # comparison function for ranking
+    def compare(a, b):
+        if a.node_id() == b.node_id():
+            return 0
+        if a.is_constant() and b.is_constant():
+            return (a.constant_value() > b.constant_value()) - (a.constant_value()\
```

Comment:
Why not simply `a.constant_value() - b.constant_value()`? Subtracting Booleans looks a bit strange...